quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/services/QuarkusCodeActions.java

```diff
@@ -0,0 +1,115 @@
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.Range;
+
+import com.redhat.quarkus.commons.ExtendedConfigDescriptionBuildItem;
+import com.redhat.quarkus.commons.QuarkusProjectInfo;
+import com.redhat.quarkus.ls.commons.BadLocationException;
+import com.redhat.quarkus.ls.commons.CodeActionFactory;
+import com.redhat.quarkus.model.PropertiesModel;
+import com.redhat.quarkus.model.PropertyKey;
+import com.redhat.quarkus.settings.QuarkusFormattingSettings;
+import com.redhat.quarkus.utils.PositionUtils;
+import com.redhat.quarkus.utils.QuarkusPropertiesUtils;
+
+/**
+ * The Quarkus code actions
+ * 
+ * @author Angelo ZERR
+ *
+ */
+class QuarkusCodeActions {
+
+	private static final float MAX_DISTANCE_DIFF_RATIO = 0.1f;
+
+	private static final Logger LOGGER = Logger.getLogger(QuarkusCodeActions.class.getName());
+
+	/**
+	 * Returns code actions for the given diagnostics of the application.properties
+	 * <code>document</code> by using the given Quarkus properties metadata
+	 * <code>projectInfo</code>.
+	 * 
+	 * @param context            the code action context
+	 * @param range              the range
+	 * @param document           the properties model.
+	 * @param projectInfo        the Quarkus properties
+	 * @param formattingSettings the formatting settings.
+	 * @return the result of the code actions.
+	 */
+	public List<CodeAction> doCodeActions(CodeActionContext context, Range range, PropertiesModel document,
+			QuarkusProjectInfo projectInfo, QuarkusFormattingSettings formattingSettings) {
+		List<CodeAction> codeActions = new ArrayList<>();
+		if (context.getDiagnostics() != null) {
+			// Loop for all diagnostics
+			for (Diagnostic diagnostic : context.getDiagnostics()) {
+				if (ValidationType.unknown.name().equals(diagnostic.getCode())) {
+					// Manage code action for unknown
+					doCodeActionsForUnknown(diagnostic, document, projectInfo, codeActions);
+				}
+			}
+		}
+		return codeActions;
+	}
+
+	/**
+	 * Creation code action for 'unknown' property by searching similar name from
+	 * the known Quarkus properties.
+	 * 
+	 * <p>
+	 * LIMITATION: mapped property are not supported.
+	 * </p>
+	 * 
+	 * @param diagnostic  the diagnostic
+	 * @param the         properties model.
+	 * @param projectInfo the Quarkus properties
+	 * @param codeActions code actions list to fill.
+	 */
+	private void doCodeActionsForUnknown(Diagnostic diagnostic, PropertiesModel document,
+			QuarkusProjectInfo projectInfo, List<CodeAction> codeActions) {
+		try {
+			// Get property name by using the diagnostic range
+			PropertyKey propertyKey = (PropertyKey) document.findNodeAt(diagnostic.getRange().getStart());
+			String propertyName = propertyKey.getPropertyName();
+			// Loop for each metata property
+			for (ExtendedConfigDescriptionBuildItem metaProperty : projectInfo.getProperties()) {
+				if (QuarkusPropertiesUtils.isMappedProperty(metaProperty.getPropertyName())) {
+					// FIXME: support mapped property

```

Comment:
Do you think we might support mapped property in the future?

Reply:
It should be fine but today I have not found a clean solution to do that.